### PR TITLE
test: refactor use of getrawmempool in functional tests for efficiency

### PR DESCRIPTION
**Pull request by mjdietzx**
I don't think this changes the intention of the test. But it does shave ~30 seconds off the time it takes to run. From what I've seen our CI `macOS 11 native [gui] [no depends]` runs `mempool_updatefrom.py` in ~135 seconds. After this PR it should run in ~105 seconds

I noticed this improvement should probably be made when testing performance/runtimes of https://github.com/bitcoin/bitcoin/pull/22698. But I wanted to separate this out from that PR so the affects of each is decoupled

Edit: The major change in this PR is improving mempool_updatefrom.py's runtime as this is a very long running test. Then made the same efficiency improvements across all the functional tests as it made since to do that here

### CODE REVIEW

File: test/functional/mempool_updatefromblock.py [hunk 1]
```diff
@@ -109,10 +109,10 @@ def transaction_graph_test(self, size, n_tx_to_mine=None, start_input_txid='', e
...
         for k, tx in enumerate(tx_id):
             self.log.debug('Check transaction #{}.'.format(k))
-            assert_equal(self.nodes[0].getrawmempool(True)[tx]['descendantcount'], size - k)
-            assert_equal(self.nodes[0].getrawmempool(True)[tx]['descendantsize'], sum(tx_size[k:size]))
-            assert_equal(self.nodes[0].getrawmempool(True)[tx]['ancestorcount'], k + 1)
-            assert_equal(self.nodes[0].getrawmempool(True)[tx]['ancestorsize'], sum(tx_size[0:(k + 1)]))
+            assert_equal(self.nodes[0].getmempoolentry(tx)['descendantcount'], size - k)
+            assert_equal(self.nodes[0].getmempoolentry(tx)['descendantsize'], sum(tx_size[k:size]))
+            assert_equal(self.nodes[0].getmempoolentry(tx)['ancestorcount'], k + 1)
+            assert_equal(self.nodes[0].getmempoolentry(tx)['ancestorsize'], sum(tx_size[0:(k + 1)]))
```
Comment (glozow) [inline test/functional/mempool_updatefromblock.py:115]: If you're going to do this to save time, you might as well just call the RPC once.
```suggestion
            entry = self.nodes[0].getmempoolentry(tx)
            assert_equal(entry['descendantcount'], size - k)
            assert_equal(entry['descendantsize'], sum(tx_size[k:size]))
            assert_equal(entry['ancestorcount'], k + 1)
            assert_equal(entry['ancestorsize'], sum(tx_size[0:(k + 1)]))
```

Comment (glozow) [reply test/functional/mempool_updatefromblock.py:115]: This puts me at 13sec (vs 33sec on master).

**Review by glozow — COMMENTED** [review]
Concept ACK, it certainly saves time to just fetch the entry we need instead of the entire mempool, particularly because this test puts a lot of transactions in there. It is significantly faster on my machine as well.


Should do this for other functional tests as well, e.g:

test/functional/mempool_compatibility.py
test/functional/rpc_fundrawtransaction.py
test/functional/mempool_package_limits.py

(From a quick grep for `getrawmempool(True)` and `getrawmempool(verbose=True)` and eliminating the ones that should actually use getrawmempool)

**Comment by mjdietzx** [comment]
Thanks @glozow - good suggestions, I'll follow up with these shortly

**Review by josibake — APPROVED** [review]
Code Review and Concept ACK

+1 for always preferring `getmempoolentry` over `getrawmempool` when you are referencing specific txns.

left a suggestion about using deprecated fee fields, but like i mentioned, it might be out of scope for this PR.

i would also suggest updating the title and description of the PR to reflect that you are globally replacing references to `getrawmempool` (right now it's still specific to `mempool_updatefrom`)

after updating the title and description, https://github.com/bitcoin/bitcoin/pull/22707/commits/86dbd54ae8a8f9c693c0ea67114bbff24a0754df and https://github.com/bitcoin/bitcoin/pull/22707/commits/77349713b189e80f2c140db4df50177353a1cb83 should be squashed

you could even squash all three, since both changes fall under refactoring for efficiency :man_shrugging:

**Comment by mjdietzx** [comment]
Thanks for the review @josibake ! Updated PR title/description accordingly

> left a suggestion about using deprecated fee fields, but like i mentioned, it might be out of scope for this PR.

I think this is/should be done in your PR #22689. If we have merge conflicts one of us will just need to rebase depending on whose PR gets merged first.

**Comment by theStack** [comment]
Concept ACK

**Review by theStack — APPROVED** [review]
Tested ACK 47c48b5f35b4910fcf87caa6e37407e67d879c80

On my machine, the speed-up of `mempool_updatefromblock.py` is quite impressive (>2x):

master branch:
```
$ time ./test/functional/mempool_updatefromblock.py
    3m18.16s real     2m04.55s user     0m33.54s system
```

PR branch:
```
$ time ./test/functional/mempool_updatefromblock.py
    1m29.11s real     0m34.37s user     0m14.79s system
```

**Review by darosior — COMMENTED** [review]
Concept ACK

**Pull request merged**